NeuroID/src/main/java/com/neuroid/tracker/service/NIDConfigurationService.kt

```diff
@@ -0,0 +1,30 @@
+package com.neuroid.tracker.service
+
+import com.neuroid.tracker.models.NIDRemoteConfig
+
+class NIDConfigurationService(
+    apiService: NIDApiService,
+    remoteConfigListener: OnRemoteConfigReceivedListener,
+    key: String
+): RetrySender() {
+
+    private val responseCallback = object: NIDResponseCallBack {
+        override fun <T> onSuccess(code: Int, response: T) {
+            remoteConfigListener.onRemoteConfigReceived(response as NIDRemoteConfig)
+        }
+
+        override fun onFailure(code: Int, message: String, isRetry: Boolean) {
+            remoteConfigListener.onRemoteConfigReceivedFailed("no can get config for key: $key, use defaults")
```

Comment:
Can we change the error message to sound a little more user friendly? Something like "Unable to retrieve the remote configuration for key $key. The default values will be used instead"

Reply:
I agree.